src/Uno.UI/UI/Xaml/UIElement.TextHelper.wasm.cs

```diff
@@ -97,6 +97,30 @@ internal void SetFontFamily(object localValue)
 			}
 		}
 
+		internal void SetFontStretch(object localValue)
+		{
+			if (localValue == DependencyProperty.UnsetValue)
+			{
+				ResetStyle("font-stretch");
+			}
+			else
+			{
+				string cssStretch = (FontStretch)localValue switch
+				{
+					FontStretch.UltraCondensed => "ultra-condensed",
+					FontStretch.ExtraCondensed => "extra-condensed",
+					FontStretch.Condensed => "condensed",
+					FontStretch.SemiCondensed => "semi-condensed",
+					FontStretch.SemiExpanded => "semi-expanded",
+					FontStretch.Expanded => "expanded",
+					FontStretch.ExtraExpanded => "extra-expanded",
+					FontStretch.UltraExpanded => "ultra-expanded",
+					_ => "normal",
+				};
+				this.SetStyle("font-stretch", cssStretch);
```

Comment:
We should not force `Undefined` to be treated as `Normal` since some font (let's say `Arial Narrow`) are not defined as "_normal - 100%_" by default.

Reply:
In that case the font should render as "narrow" automatically, by its definition.

Reply:
WinUI has this code in TextBlock.cpp:


```cpp
DWRITE_FONT_STRETCH dwriteFontStretch = MIN(MAX(DWRITE_FONT_STRETCH_ULTRA_CONDENSED, static_cast<DWRITE_FONT_STRETCH>(pTextFormatting->m_nFontStretch)), DWRITE_FONT_STRETCH_ULTRA_EXPANDED);
```

Looks like they may be forcing an in-range value. So, with the above logic from WinUI, undefined will become Ultra condensed. That also matches what I visually see when I try on WinUI.

![image](https://github.com/unoplatform/uno/assets/31348972/54d2050f-35ec-4c66-92fc-a84d608c69ab)

```xml
<TextBlock FontFamily="Arial" FontStretch="Undefined" Text="Hello world (undefined)" />
<TextBlock FontFamily="Arial" FontStretch="Normal" Text="Hello world (Normal)" />
<TextBlock FontFamily="Arial" FontStretch="UltraCondensed" Text="Hello world (ultra condensed)" />
<TextBlock FontFamily="Arial" FontStretch="UltraExpanded" Text="Hello world (ultra expanded)" />
```